client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { FC, useEffect, useState } from 'react';
+import { StandardTextFieldProps } from '@common/components';
+import { BasicTextInput } from './BasicTextInput';
+import { NumUtils, RegexUtils } from '@common/utils';
+import { useFormatNumber, useCurrency } from '@common/intl';
+export interface NumericTextInputProps
+  extends Omit<StandardTextFieldProps, 'onChange'> {
+  onChange?: (value: number | undefined) => void;
+  width?: number | string;
+  defaultValue?: number;
+  allowNegative?: boolean;
+  min?: number;
+  max?: number;
+  decimalLimit?: number;
+  step?: number;
+  multiplier?: number;
+  value?: number | undefined;
+}
+
+export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
+  (
+    {
+      sx,
+      InputProps,
+      width = 75,
+      onChange = () => {},
+      defaultValue,
+      allowNegative,
+      min = allowNegative ? -NumUtils.MAX_SAFE_API_INTEGER : 0,
+      max = NumUtils.MAX_SAFE_API_INTEGER,
+      decimalLimit = 0,
+      step = 1,
+      multiplier = 10,
+      value,
+      ...props
+    },
+    ref
+  ) => {
+    const { format, parse } = useFormatNumber();
+    const {
+      options: { separator, decimal },
+    } = useCurrency();
+    const [textValue, setTextValue] = useState(format(value ?? defaultValue));
+
+    useEffect(() => {
+      setTextValue(format(value));
+    }, [value]);
```

Comment:
This is one "rule" I often see that I don't really understand -- it often seems to cause more problems than it solves. In this particular case, `format` is a _static_ method returned from `useFormatNumber` and it loads before the `useEffect` is first run. We know it can't be changed, so why do we need to include it in the dependencies?

And in fact, in this case, it actually breaks the functionality -- the `useEffect` runs twice after each change, with one of them getting an `undefined` value -- the outcome is that the typed decimal input (before subsequent digits are added) gets immediately wiped out, so can't add a decimal.

Do you know what the reason for this "rule" is, as I think the dependency array should be as small as possible to limit re-renders, and putting in things that we know won't change seems wasteful?

Reply:
if the function executed by useEffect is using functions or variables and those change, but aren't part of the dependencies, then the effect isn't re-run with the modified values. that's what the linter is trying to catch. and it's a valid concern.

I've found that blindly adding in all dependencies to calm the linter has caused problems though, with the effect being called more than expected. In this case it's good to be explicit and to comment why the deps array is limited. If nothing else, adding the linter option has provided motivation for thinking more carefully about dependencies. I think it's good to have the discussion

Reply:
Okay @mark-prins , have added comment for now: 4a7596d15

> I've found that blindly adding in all dependencies to calm the linter has caused problems though, with the effect being called more than expected.

Yes, that's exactly what was happening.